src/Controls/src/Core/Handlers/NavigationPage/NavigationPageHandler.Android.cs

```diff
@@ -125,254 +130,24 @@ void OnPushed(object sender, NavigationRequestedEventArgs e)
 
 		internal void OnPop()
 		{
-			NavHost.NavController.NavigateUp();
+			VirtualViewWithValidation()
```

Comment:
Fire everything through the Navigation page

Life's easier when you always follow the same circle